arch/arm/core/aarch32/cortex_m/debug.c

```diff
@@ -42,6 +42,7 @@ bool z_arm_debug_monitor_event_error_check(void)
 		 * would mean the system would be stuck in an infinite loop).
 		 */
 		return true;
+	} else {
```

Comment:
FWIW, MISRA says:

> A final else statement shall always be provided whenever an if statement is followed by a sequence of one or more else if constructs. **The else statement shall contain at least either one side effect or a comment.** *(emphasis added)*

so this doesn't actually address the violation in full.

Reply:
Thank you @pabigot ! I wondered about that. I'll fix it them to fully address the issue :)